erts/emulator/beam/jit/asm_load.c

```diff
@@ -859,6 +873,58 @@ static const BeamCodeLineTab *finish_line_table(LoaderState *stp,
     return line_tab_ro;
 }
 
+static const BeamDebugTab *finish_debug_table(LoaderState *stp,
+                                              char *module_base,
+                                              size_t module_size) {
+    BeamFile_DebugTable *debug = &stp->beam.debug;
+    const BeamDebugTab *debug_tab_ro;
+    byte *debug_tab_rw_base;
+    BeamDebugTab *debug_tab_top;
+    Eterm *debug_tab_terms;
+    BeamDebugItem *debug_tab_items;
+    Uint item_count = (Uint)debug->item_count;
+    Uint term_count = (Uint)debug->term_count;
+    Uint i;
+
+    if (item_count == 0) {
+        return NULL;
+    }
+
+    debug_tab_ro = (const BeamDebugTab *)beamasm_get_rodata(stp->ba, "debug");
+    debug_tab_rw_base = get_writable_ptr(stp->executable_region,
+                                         stp->writable_region,
+                                         debug_tab_ro);
+    debug_tab_top = (BeamDebugTab *)debug_tab_rw_base;
+    debug_tab_terms = (Eterm *)(debug_tab_top + 1);
+    debug_tab_items = (BeamDebugItem *)(debug_tab_terms + term_count);
```

Comment:
Nitpick:

```suggestion
    debug_tab_terms = (Eterm *)&debug_tab_top[1];
    debug_tab_items = (BeamDebugItem *)&debug_tab_terms[term_count];
```

For this to work we should also assert that the alignment of `BeamDebugTab` is at least that of `Eterm`, and that the alignment of `BeamDebugItem` is at most that of `Eterm`, but `_Alignof` requires C11. :-|